Sources/AmplitudeCore/Remote Config/RemoteConfigClient.swift

```diff
@@ -0,0 +1,378 @@
+                  logger: logger,
+                  storage: RemoteConfigUserDefaultsStorage(instanceName: instanceName))
+    }
+
+    init(apiKey: String,
+         serverUrl: String,
+         logger: Logger = ConsoleLogger(),
+         storage: Storage = RemoteConfigUserDefaultsStorage(),
+         urlSessionConfiguration: URLSessionConfiguration = .ephemeral) {
+        self.apiKey = apiKey
+        self.serverUrl = serverUrl
+        self.logger = logger
+        self.storage = storage
+        self.urlSession = URLSession(configuration: urlSessionConfiguration)
+
+        fetchLocalTask = Task {
+            guard let config = try await storage.fetchConfig() else {
+                throw RemoteConfigError.notInCache
+            }
+            return config
+        }
+
+        var futureSelf: RemoteConfigClient? = nil
+        fetchRemoteTask = Task {
+            guard let futureSelf else {
+                throw RemoteConfigError.badResponse
+            }
+            let configInfo = try await futureSelf.fetch()
+            try? await storage.setConfig(configInfo)
+            return configInfo
+        }
+
+        super.init()
+
+        futureSelf = self
+    }
+
+    /**
+     Subscribe for updates to remote config. Callback is guaranteed to be called at least once, whether we are able to fetch a config or not.
+
+     - Parameters:
+        - key: A String containing a series of period delimited keys to filter the returned config. Ie, {a: {b: {c: ...}}} would return {b: {c: ...} for "a" or {c: ...} for "a.b"
+        - deliveryMode: How the initial callback is sent. See ``RemoteConfigClient/DeliveryMode`` for more details.
+        - callback: A block that will
+     - Returns: A token that can be used to unsubscribe from updates
+     */
+    @discardableResult
+    public nonisolated func subscribe(key: String? = nil,
+                                      deliveryMode: DeliveryMode = .all,
+                                      callback: @escaping @Sendable RemoteConfigCallback) -> Any {
+        let id = UUID()
+        Task {
+            await _subscribe(id: id, key: key, deliveryMode: deliveryMode, callback: callback)
+        }
+        return id
+    }
+
+    private func _subscribe(id: UUID,
+                            key: String?,
+                            deliveryMode: DeliveryMode,
+                            callback: @escaping @Sendable RemoteConfigCallback) {
+        Task {
+            let callbackInfo = CallbackInfo(id: id, key: key, deliveryMode: deliveryMode, callback: callback)
+            callbacks.append(callbackInfo)
+
+            switch callbackInfo.deliveryMode {
+            case .all:
+                Task {
+                    await withThrowingTaskGroup(of: (configInfo: RemoteConfigInfo, source: Source).self) { [fetchLocalTask, fetchRemoteTask] taskGroup in
+                        // send remote first, if it's already complete we can skip the cached response
```

Comment:
Personal comment: remote is always fetched. In other words
- if remote is fetched first, no cache fetch
- if cache is fetched first, still fetching remote